x/nitro/replay/replay.go

```diff
@@ -0,0 +1,187 @@
+//go:build !codeanalysis
+// +build !codeanalysis
+
+package replay
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/sei-protocol/sei-chain/utils"
+	"github.com/sei-protocol/sei-chain/x/nitro/types"
+)
+
+func Replay(ctx sdk.Context, txs [][]byte, accounts []types.Account, programs []types.Account) ([]types.Account, error) {
+	// there can be at most one replay per Sei block
+	inputDirectory := fmt.Sprintf("/tmp/replay_input_%d/", ctx.BlockHeight())
+	if err := os.Mkdir(inputDirectory, os.ModePerm); err != nil {
```

Comment:
will the replay cause the validator node to fall behind as it requires many os operations